api/subscriptions/permissions.py

```diff
@@ -0,0 +1,16 @@
+from rest_framework import permissions
+
+from osf.models.notifications import NotificationSubscription
+
+
+class IsOwner(permissions.BasePermission):
```

Comment:
This should likely be checking permissions on the related object not on the subscription itself. Matt suggested this an may have better recommendations on how to implement it properly.

Reply:
If we end up keeping IsOwner versus a different way to do the permission check, it should probably be "IsSubscriptionOwner" to be clear what is owned.